pkg/operator/ceph/config/config.go

```diff
@@ -0,0 +1,199 @@
+			configOrder: []string{},
+		}
+		c.sections[name] = s
+		c.sectionOrder = append(c.sectionOrder, name)
+	}
+	return s
+}
+
+// Set sets the Ceph config key to the given value. Set accepts Ceph keys in both
+// space-separated-words and underscore-separated-words formats. Set modifies the section in-place
+// and also returns the same modified-in-place Section to give flexibility of usage.
+//
+// NOTE that if a config item in a map should have quotes around it, they will have to be added to
+// the string explicitly. e.g., a Value of " test " will be interpreted as as 'Key = test', whereas
+// a Value of "\" test \"" will be interpreted as 'Key = " test "'
+func (s *Section) Set(key, value string) *Section {
+	key = normalizeKey(key)
+	_, ok := s.configs[key]
+	if !ok {
+		s.configOrder = append(s.configOrder, key)
+	}
+	s.configs[key] = value
+	return s
+}
+
+// Merge modifies the Section in-place adding values from the overrides Section and overriding
+// existing values with the new value in overrides.
+func (s *Section) Merge(overrides *Section) {
+	for k, v := range overrides.configs {
+		s.Set(k, v)
+	}
+}
+
+// Merge modifies the Config in-place, adding Sections from the overrides config, and overriding
+// existing Sections with Section values from overrides.
+func (c *Config) Merge(overrides *Config) {
+	for hdr, sec := range overrides.sections {
+		c.Section(hdr).Merge(sec)
+	}
+}
+
+// FileText converts the Ceph Config to a string formatted to look like the text of a Ceph config
+// file. If errors are encountered, the error is returned with an empty string.
+// Double newlines after headers and at the end of a section; single newline betw header and section
+func (c *Config) FileText() (string, error) {
+	f := ini.Empty()
+	for _, hdr := range c.sectionOrder {
+		sec, ok := c.sections[hdr]
+		if !ok {
+			return "", fmt.Errorf("Error: section header %s not found: %+v", hdr, c)
+		}
+		s, err := f.NewSection(hdr)
+		if err != nil {
+			return "", fmt.Errorf("Error converting config to ini file text: %+v. %+v", c, err)
+		}
+		for _, k := range sec.configOrder {
+			v, ok := sec.configs[k]
+			if !ok {
+				return "", fmt.Errorf("Error: config %s not found in section %s: %+v", k, hdr, sec)
+			}
+			if _, err := s.NewKey(k, v); err != nil {
+				return "", fmt.Errorf("Error converting config to ini file text: %+v. %+v", c, err)
+			}
+		}
+	}
+	b := new(bytes.Buffer)
+	if _, err := f.WriteTo(b); err != nil {
+		return "", fmt.Errorf("Error converting config to ini file text: %+v. %+v", c, err)
+	}
+	return b.String(), nil
+}
+
+// NewFlag returns the key-value pair in the format of a Ceph command line-compatible flag.
+func NewFlag(key, value string) string {
+	// A flag is a normalized key with underscores replaced by dashes.
+	// "debug default" ~normalize~> "debug_default" ~to~flag~> "debug-default"
+	n := normalizeKey(key)
+	f := strings.Replace(n, "_", "-", -1)
+	return fmt.Sprintf("--%s=%s", f, value)
```

Comment:
do we need to put quotes around the value in case there is a space? Or will the pod spec handle this ok already as the string slince?

Reply:
I can't find any documentation that confirms either way. I didn't think any Ceph args could have spaces and that this would still be safe, but @liewegas might have better thoughts. I have a worry about what this might do if args are specified with quotes in them like `--debug-prefix="debug "`. We know that it is specified as `"debug "`, but a user might override with `'my text - '`. 

I guess we could always add `"` quotes and if the string already has `"` quotes add escape sequences to them. I'm probably thinking about this too hard, but I'm also aware that I don't know what kinds of issues we might see with doing that kind of thing.

Reply:
Note to self: Add `--debug-prefix="debug "` as a commandline arg. Also `--log-to-stderr=true` and --cluster-log-to-stderr=true`